lib/rules/space-in-parens.js

```diff
@@ -184,9 +188,9 @@ module.exports = function(context) {
 
     return {
         "Program": function checkParenSpaces(node) {
-            var tokens = node.tokens;
-            var prevToken, nextToken;
+            var tokens, prevToken, nextToken;
             exceptions = getExceptions();
+            tokens = sourceCode.tokensAndComments;
```

Comment:
@nzakas Unless I am missing something, you are talking about the extra cost of the function call, right? And that would only be paid when such function is called (right now, when this rule is enabled).

On the other hand, making it a property makes that code be executed every time `SourceCode` is instantiated. I guess that cost will be much higher compared to a function call.